### PR TITLE
GH-2259 changes to PR template, guidelines, and activation of formatting plugins

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2259 

Briefly describe the changes proposed in this PR:

- xml formatting check part of automatic validation by CI
- xml formatting format part of automatic formatting as part of local build
- updated PR template to add links to guidelines and point out formatting conventions
- made it a bit clearer which maven goals you can run to make this easier

---- 
PR Author Checklist: 

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [ ] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,10 +6,11 @@ Briefly describe the changes proposed in this PR:
 <!-- short description of your change goes here -->
 
 ---- 
-PR Author Checklist:
```
Comment (hmottestad) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Could you add the maven command for formatting somewhere on this template as a comment?

**Comment by hmottestad** [comment]
Other than my comment. Looks good.

**Comment by jeenbroekstra** [comment]
> Other than my comment. Looks good.

Sorry, which comment is that?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hmottestad) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: > > Other than my comment. Looks good.
> 
> Sorry, which comment is that?

GitHub had made it a draft comment. Sorry for the confusion. Fixed it now.

**Pull request merged**